### DISCUSSION TITLE
Trigger a config reload

### DISCUSSION THREAD

**Discussion by agschaid** [wezterm/wezterm#1461] [Q&A] [answered]
Hi everybody.

I am wondering whether there is a way to trigger a config reload of wezterms config from a script.

Here is my specific issue: I have a setup where I can toggle large parts of my system UI between Solarized Dark and Light. The core of this setup is the `~/.uicolor` that is hold "state".

In my wezterm config I currently have this function (it's probably ugly. I've "learned" Lua about 30 minutes ago):

```
function color()
  for line in io.lines("/home/agl/.uicolor") do
    if line == "light" then
      return "Builtin Solarized Light"
    end
  end

  return "Builtin Solarized Dark"
end
```

This works well for new instances of wezterm but not running ones (since `~/.uicolor` is of course not being watched). So I'd need/want a way to hint wezterm "manually" that config reload for all instances would be in order.
Is there an official way or hack to do this?

I am aware of [this answer](https://github.com/wez/wezterm/discussions/914#discussioncomment-960983) by @wez on how to  add a custom module searcher. However if I read the documentation correctly that would only work for Lua and C files. Not for random config files.

**Comment by bew** [comment, marked as answer]
Hello,
The function that was added at the time is https://wezfurlong.org/wezterm/config/lua/wezterm/add_to_config_reload_watch_list.html and it's not only specific to C or Lua, never tried it but it seems to be generic from that documentation.

**Reply by agschaid** [reply]
Oh. great! 
Now that I see it I don't even know how I was able to miss it!
Thank you very much to point that out.

Just to be clear:
With "the documentation" I wasn't referring to the documentation of  that function but to [adding package searchers](https://www.lua.org/manual/5.3/manual.html#pdf-package.searchers).

**Reply by agschaid** [reply]
once again: thank you.

Just tried it. Works like a charm.

**Reply by agschaid** [reply]
sorry. Me again.
One tiny issue that isn't really surprising but might be of interest for future readers:
the file in question needs to exist when it is added to the watch list. The underlying system can obviously monitor changes and also the deletion of the file. But not its creation.

A minor edge case for me. Still happy ;)
